.github/workflows/pr_push_workflow.yml

```diff
@@ -0,0 +1,58 @@
+name: PR/push workflow
+
+on: [push, pull_request]
+
+jobs:
+    Fast_build:
+        # uses: oneapi-src/unified-memory-framework/.github/workflows/fast_build.yml@main
+        name: Fast build
+        strategy:
+            matrix:
+                os: [ubuntu-latest, windows-latest]
+                include:
+                    - os: ubuntu-latest
+                      nproc: $(nproc)
+                      pool_scalable: 'ON'
+                    - os: windows-latest
+                      nproc: $Env:NUMBER_OF_PROCESSORS
+                      pool_scalable: 'OFF'
+        runs-on: ${{matrix.os}}
+
+        steps:
+        - name: Checkout repository
+          uses: actions/checkout@v4
+
+        - name: Install apt packages
+          if: matrix.os == 'ubuntu-latest'
+          run: |
+            sudo apt-get update
+            sudo apt-get install -y cmake clang libnuma-dev libjemalloc-dev libtbb-dev
+
+        - name: Install pip packages
+          run: python3 -m pip install -r third_party/requirements.txt
+
+        - name: Configure CMake
+          run: >
+            cmake
+            -B ${{github.workspace}}/build
+            -DCMAKE_BUILD_TYPE=Release
+            -DUMF_FORMAT_CODE_STYLE=ON
+            -DUMF_DEVELOPER_MODE=ON
+            -DUMF_ENABLE_POOL_TRACKING=ON
+            -DUMF_BUILD_LIBUMF_POOL_SCALABLE=${{matrix.pool_scalable}}
+
+        - name: Build
+          run: >
+            cmake --build ${{github.workspace}}/build --config Release -j ${{matrix.nproc}}
+    CodeQL:
```

Comment:
Given how often the `Build` workflow fails contrary to the `CodeQL` workflow, I would require to finish the `Build` first, then run `CodeQL`. This way we could fail-fast the tests in CI and start the `Benchmark` job in parallel to `CodeQL`.

Reply:
I moved it to the "benchmark" level. Order etc. can be considered later.